scripts/build_shapes.py

```diff
@@ -120,10 +128,51 @@ def get_GADM_layer(country_list, layer_id, geo_crs, update=False, outlogging=Fal
             three_2_two_digits_country(twoD_c) for twoD_c in geodf_temp["GID_0"]
         ]
 
+        # GID_0 may have some exotic values, "COUNTRY" column may be used instead
+        geodf_temp["GID_0"] = geodf_temp.apply(
+            lambda x: restore_country_code_by_name(x, country_code=country_code), axis=1
+        )
+
+        # "not found" hardcoded according to country_converter conventions
+        geodf_temp.drop(
+            geodf_temp[geodf_temp["GID_0"] == "not found"].index, inplace=True
+        )
+
         # create a subindex column that is useful
         # in the GADM processing of sub-national zones
         geodf_temp["GADM_ID"] = geodf_temp[f"GID_{layer_id}"]
 
+        if layer_id >= 1:
+            available_gadm_codes = geodf_temp["GADM_ID"].unique()
+            code_three_digits = two_2_three_digits_country(country_code)
+
+            # normally the GADM code starts the ISO3
+            non_std_gadm_codes = [
+                w for w in available_gadm_codes if not w.startswith(code_three_digits)
+            ]
+
+            if len(non_std_gadm_codes) > 0:
+
+                # a "minimalistic" approach outputs a single file (~200kb for CN + IN)
+                d = {
+                    "country_code": country_code,
+                    "non_standard_gadm_codes": non_std_gadm_codes,
+                }
+                df = pd.DataFrame(data=d)
+                output_path = "non_standard_gadm_codes.csv"
+                df.to_csv(
+                    output_path,
```

Comment:
It would be better to keep only the non-appending mode below.
For consecutive runs the appending approach may add redundant lines.

Reply:
Agree, revised

As discussed, there is also a need to add a config parameter and a snakemake output if we'll decide to keep it in a cleaned version. I'd suggest to think if it's really worth to have this output in the mainstream workflow: wouldn't it be a bit of over-complication?